File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: list[int]) -> int:
        seen = set()
        total = 0
        has_non_negative = False

        for num in nums:
            if num >= 0:
                has_non_negative = True
                if num not in seen:
                    seen.add(num)
                    total += num

        if has_non_negative:
            return total
        else:
            # All numbers are negative
            return max(nums)

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.go
func maxSum(nums []int) int {
    seen := make(map[int]bool)
    sum := 0
    hasNonNegative := false

    for _, num := range nums {
        if num >= 0 {
            hasNonNegative = true
            if !seen[num] {
                seen[num] = true
                sum += num
            }
        }
    }

    if hasNonNegative {
        return sum
    }

    // If all numbers are negative, return the maximum one
    maxNegative := nums[0]
    for _, num := range nums {
        if num > maxNegative {
            maxNegative = num
        }
    }

    return maxNegative
}